sphinx/themes/basic/static/doctools.js

```diff
@@ -294,21 +294,28 @@ var Documentation = {
       if (activeElementType !== 'TEXTAREA' && activeElementType !== 'INPUT' && activeElementType !== 'SELECT'
           && activeElementType !== 'BUTTON' && !event.altKey && !event.ctrlKey && !event.metaKey
           && !event.shiftKey) {
-        switch (event.keyCode) {
-          case 37: // left
+        switch (event.key) {
+          case 'ArrowLeft':
             var prevHref = $('link[rel="prev"]').prop('href');
             if (prevHref) {
               window.location.href = prevHref;
               return false;
             }
-            break;
-          case 39: // right
+          case 'ArrowRight':
             var nextHref = $('link[rel="next"]').prop('href');
             if (nextHref) {
               window.location.href = nextHref;
               return false;
             }
-            break;
+          case '/':
+          case 's':
+            $('input[name=q]').focus();
+            return false;
+          case 'h':
+            newurl = window.location.href.replace(new RegExp('\\?highlight=[^#]*'), '');
+            history.pushState({}, null, newurl);
```

Comment:
Thanks for the update, this works really well, without re-loading the page!

Would it be possible to move this into the `hideSearchWords()` function?

This way, the URL would also be modified when clicking the "hide search matches" link, which I think would be great!

This could even be a separate (less contentious) PR, because it really doesn't have anything to do with shotcuts.

Reply:
I welcome the idea and I've pushed a commit that does that.

Reply:
> This could even be a separate (less contentious) PR, because it really doesn't have anything to do with shortcuts.

I've just created such a PR: #9551.